langfuse/model.py

```diff
@@ -184,6 +219,21 @@ def compile(self, **kwargs) -> List[ChatMessageDict]:
             for chat_message in self.prompt
         ]
 
+    def variable_names(self) -> List[ChatMessageVariables]:
```

Comment:
I think the utility of getting a list of variable names of a prompt it to verify before rendering the template whether all vars have been passed. From that lens, I don't understand why having a list of `ChatMessageVariables` has utility for chat prompts rather than a simple list of strings with the variable names. Is there a use case for you here @rhighs ?

Reply:
Nope, not really. I did it this way to be consistent with the return type you've been using in the compile method above. But at this point I think you're right, a simple list is enough.

Reply:
A reason could be that if someone's using a ChatPromptTemplate he needs some order of organization in his prompts. Variable names in my case are exceptionally useful for post processing purposes, e.g. i know exactly what is the prompt i want to process and then work on it's variables. To me it makes sense having this result organized in this way, otherwise I'd have to manipulate the output in reduction
```python
functools.reduce(operator.iconcat, [
            TemplateParser.find_variable_names(chat_message["content"])
            for chat_message in self.prompt
        ], [])
```